nemoguardrails/library/privateai/request.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for handling Private AI detection requests."""
+
+import json
+import logging
+from typing import Any, Dict, List, Optional
+
+import aiohttp
+
+log = logging.getLogger(__name__)
+
+
+async def private_ai_detection_request(
+    text: str,
+    enabled_entities: List[str],
+    server_endpoint: str,
+    api_key: Optional[str] = None,
+):
+    """
+    Send a detection request to the Private AI API.
+
+    Args:
+        text (str): The text to analyze.
+        enabled_entities (List[str]): List of entity types to detect.
+        server_endpoint (str): The API endpoint URL.
+        api_key (Optional[str]): The API key for the Private AI service.
+
+    Returns:
+        bool or None: True if entities detected, False if not, None if request failed.
```

Comment:
Instead of returning None, can we raise an Exception when the request fails? I think that is a better way to handle failures.